tempo_embeddings/embeddings/weaviate_database.py

```diff
@@ -0,0 +1,278 @@
+                self.model.batch_size = self.batch_size
+                self.embedding_function = wvc.config.Configure.Vectorizer.none()
+            except KeyError:
+                logger.error("If the type is 'custom_model' or 'default' you should pass the model object under Key 'model'")
+        else:
+            raise ValueError(f"Malformed embedder_config {self.embedder_config}. Check that 'type', 'api_key' and 'model' keys are properly populated.")
+
+        if os.path.exists(self.db_path):
+            self._load_config()
+        else:
+            os.makedirs(self.db_path)
+            self.config = {
+                "embedder_name": self.embedder_name,
+                "embedder_type": self.embedder_config["type"],
+                "existing_collections": [],
+            }
+        
+        
+    def connect(self):
+        with weaviate.connect_to_local() as client:
+            # Do something with the client
+            logger.info(type(client))
+            logger.info("Weaviate Server Is Up: %s", client.is_ready())
+            return client.is_ready()
+
+
+    def _save_config(self):
+        if self.config:
+            with open(f"{self.db_path}/config.json", "w", encoding="utf-8") as f:
+                json.dump(self.config, f, indent=2)
+    
+
+    def _load_config(self):
+        with open(f"{self.db_path}/config.json", encoding="utf-8") as f:
+            self.config = json.load(f)
+
+
+    def get_available_collections(self):
+        if self.config:
+            return self.config.get("existing_collections", [])
+        return []
+
+
+    def create_new_collection(
+        self,
+        name: str,
+        corpus: Corpus = None,
+        # collection_metadata = None, # This Dict can be used in the future to give a precise SCHEMA. NOT IMPLEMENTED YET!
+    ) -> None:
+
+        # Create NEW collection and Embeds the given passages. Do nothing otherwise
+        if name in self.config["existing_collections"]:
+            raise ValueError("This collection has been created already! Try 'insert_corpus()' if you want to add more items to the collection")
+        
+        # If the collection is new then insert the corresponding passages already
+        if corpus:
+            self.insert_corpus(name, corpus)
+            self.config["existing_collections"].append(name)
+            self._save_config()
+            logger.info("Created NEW collection '%s'", name)
+        
+
+    def delete_collection(self, name):
+        with weaviate.connect_to_local() as client:
+            try:
+                client.collections.delete(name)
+                self.config["existing_collections"] = [c for c in self.config["existing_collections"] if c != name]
+                self._save_config()
+                logger.info("Succesfully deleted %s", name)
+            except Exception as e:
+                logger.warning("delete_collection() caused exception %s", e)
```

Comment:
This should probably be logged as an error:
```suggestion
                logger.error("delete_collection() caused exception %s", e)
```